components/server/src/api/rate-limited.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { IRateLimiterOptions } from "rate-limiter-flexible";
+
+const RATE_LIMIT_METADATA_KEY = Symbol("RateLimited");
+
+export function RateLimited(options: IRateLimiterOptions) {
+    return Reflect.metadata(RATE_LIMIT_METADATA_KEY, options);
+}
+
+export namespace RateLimited {
+    const defaultOptions: IRateLimiterOptions = {
+        points: 200,
+        duration: 60,
```

Comment:
let's add the unit, i.e. `durationSec` or so.

Reply:
ah, I see it is not ours 😢

Reply:
it is type from library, we will introduce indirection if we would like to some other properties